### DISCUSSION TITLE
How to log when response body length is less than 1k

### DISCUSSION THREAD

**Discussion by gaoxingliang** [apache/apisix#8750] [Q&A]
I want to log the response body in file-logger and clickhouse-logger.
but I don't know how to limit the response body length.
I tried with below ideas:
1.   register a trunked string in nginx conf.
```
 body_filter_by_lua 'local resp_body_trunc = string.sub(ngx.arg[1], 1, 1000);ngx.ctx.buffered = (ngx.ctx.buffered or "") .. resp_body_trunc;if ngx.arg[2] then;ngx.var.resp_body_trunc = ngx.ctx.buffered;end';


```
The original lua code is:
```
body_filter_by_lua '
        local resp_body = string.sub(ngx.arg[1], 1, 1000)
        ngx.ctx.buffered = (ngx.ctx.buffered or "") .. resp_body
        if ngx.arg[2] then
          ngx.var.resp_body = ngx.ctx.buffered
        end
        ';
```
but I don't know how to do that in apisix yaml config.

2. use the lua resty filter language:

below using a variable named `body_bytes_sent`. I don;t know what's the correct variable. some variables from here - https://apisix.apache.org/docs/apisix/plugins/file-logger/
```
{
    "path": "/usr/local/apisix/logs/apisix.log",
    "include_resp_body": true,
    "include_resp_body_expr":[
        [
            "body_bytes_sent",
            "<",
            1000
        ]
    ]

}
```
**Any idea to NOT log the response body if it's too large？**

**Comment by MekelCon** [comment]
Hello,

list of available variable are here :
http://nginx.org/en/docs/varindex.html

**Comment by tokers** [comment]
If your client doesn't use the chunked way to receive response body, you can check the response body size via the `Content-Length` header.

**Reply by gaoxingliang** [reply]
Hi @tokers , I'm using the chunked version. and I used the variable `body_bytes_sent` in the `include_resp_body_expr`
but it's not working.

**Reply by tokers** [reply]
`body_bytes_sent` is a variable you can know after the whole response body was sent, while in this logger plugin, the variable will be used in the `body_filter` phase; in such a case, the body is transmitting, and the total size is unknown. So it doesn't work.

**Reply by gaoxingliang** [reply]
So any idea to do this?

**Reply by tokers** [reply]
No way to do this if you use the chunk mechanism.

**Reply by gaoxingliang** [reply]
From this graph, in the log phase, it can get the total body length stuff:
![image](https://user-images.githubusercontent.com/3148215/218364956-f319c97a-1610-4778-bc31-5eb5678a89a1.png)
Maybe we can get some workarounds?
